test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -133,23 +142,28 @@ public String getID() {
         // generate Pravega Spec.
         final Map<String, Object> pravegaPersistentVolumeSpec = getPersistentVolumeClaimSpec("20Gi", "standard");
 
+        String pravegaImg = DOCKER_REGISTRY + PREFIX + "/" + PRAVEGA_IMAGE_NAME;
+
         final Map<String, Object> pravegaSpec = ImmutableMap.<String, Object>builder().put("controllerReplicas", controllerCount)
-                                                                                      .put("segmentStoreReplicas", segmentStoreCount)
-                                                                                      .put("debugLogging", true)
-                                                                                      .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
-                                                                                      //.put("options", props.getProperties())
-                                                                                      .put("options", props)
-                                                                                      .put("image",
-                                                                                       getImageSpec(DOCKER_REGISTRY + PREFIX + "/" + PRAVEGA_IMAGE_NAME, PRAVEGA_VERSION))
-                                                                                      .put("tier2", tier2Spec())
-                                                                                      .build();
+                                                                                    .put("segmentStoreReplicas", segmentStoreCount)
+                                                                                    .put("debugLogging", true)
+                                                                                    .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
+                                                                                    .put("controllerResources", getResources("2000m", "3Gi", "1000m", "1Gi"))
+                                                                                    .put("segmentStoreResources", getResources("2000m", "5Gi", "1000m", "3Gi"))
```

Comment:
Why is this necessary for using operator 4.0?

Reply:
This change is also needed for the previous version of the Operator, but it hasn't been introduced yet. In fact, we have observed system tests that induce high load in the system to fail due to lack of resources, so we need to start specifying appropriate resources for pods to avoid these failures. While this is not necessary for the Operator to work, it is necessary for the correct execution of our tests, so I'm in favor to get these changes in as soon as possible.